tests/forage_tests.cpp

```diff
@@ -0,0 +1,187 @@
+#include "catch/catch.hpp"
+#include "game.h"
+#include "player.h"
+#include "activity_handlers.h"
+#include "itype.h"
+#include "map.h"
+#include "map_helpers.h"
+#include "overmapbuffer.h"
+#include "iexamine.h"
+#include "test_statistics.h"
+
+struct forage_test_data {
+    statistics<int> calories;
+    statistics<bool> forests;
+};
+
+// returns the calories a player would harvest from an underbrush
+int forage_calories( player &p, items_location loc )
+{
+    int calories = 0;
+    const std::vector<item *> drops = find_forage_items( &p, loc );
+    for( const auto &item : drops ) {
+        if( item->is_comestible() ) {
+            calories += item->type->comestible->get_calories() * item->charges;
+        }
+    }
+    i_clear_adjacent( p.pos() );
+    return calories;
+}
+
+// gets all harvest information from an overmap tile
+std::map<harvest_id, int> get_map_terrain_harvest( const tripoint &p )
+{
+    std::map<harvest_id, int> harvests;
+    tripoint t = p;
+    for( int x = 0; x < SEEX * 2; ++x ) {
+        for( int y = 0; y < SEEY * 2; ++y ) {
+            t.x = p.x + x;
+            t.y = p.y + y;
+            harvests[ g->m.get_harvest( t ) ]++;
+        }
+    }
+    return harvests;
+}
+
+// gives a vector of tripoints with harvestable terrain or furniture
+std::vector<tripoint> get_harvest_pos( const tripoint &p )
+{
+    std::vector<tripoint> harvests;
+    tripoint t = p;
+    for( int x = 0; x < SEEX * 2; ++x ) {
+        for( int y = 0; y < SEEY * 2; ++y ) {
+            t.x = p.x + x;
+            t.y = p.y + y;
+            if( g->m.get_harvest( t ) != harvest_id( "null" ) ) {
+                harvests.push_back( t );
+            }
+        }
+    }
+    return harvests;
+}
+
+/**
+* gives a vector of tripoints for underbrush.
+* since underbrush is not a "harvestable terrain or furniture" it needs a seperate function
+* @TODO: Calculate time spent using these tripoints
+*/
+std::vector<tripoint> get_underbrush_pos( const tripoint &p )
+{
+    std::vector<tripoint> underbrush;
+    tripoint t = p;
+    for( int x = 0; x < SEEX * 2; ++x ) {
+        for( int y = 0; y < SEEY * 2; ++y ) {
+            t.x = p.x + x;
+            t.y = p.y + y;
```

Comment:
Suggest `tripoint t = p + point(x, y)`; no need for `t` to be declared outside the loop.

(Similarly above)